### ISSUE TITLE
Python Feather Breaks on Files with More Than 268,434,943 rows

### ISSUE THREAD

**Issue by jhskim89** [wesm/feather#321]
I am trying to save large files with 100MM's of rows as feather. 
But when a file has more than 268,434,943 rows, the data seems to become corrupted. 
Please see below as an example:
I created a random dataframe with 400MM rows `df_orig`. Then, I wrote it as a feather file and re-read it as a dataframe `df_copy`
```python
df = pd.DataFrame(np.random.randint(0,100,size=(400000000, 1)), columns=list('A'))
df_orig = df.reset_index(drop=False).rename(columns={"index":"base"})
df_orig['twice']=df_orig.base*2
df_orig['triple']=df_orig.base*3
feather.write_dataframe(df_orig, './test.feather')
df_copy = feather.read_dataframe('./test.feather')
```
Below is the results i get when I print out the 268,434,943th and 268,434,944th index from `df_orig` 
```python
print(df_orig.ix[268434943,:])
print("--------------------------")
print(df_orig.ix[268434944,:])
```
```
base       268434943
A                 78
twice      536869886
tripple    805304829
Name: 268434943, dtype: int64
--------------------------
base       268434944
A                 83
twice      536869888
tripple    805304832
Name: 268434944, dtype: int64
```
But when i perform the same function to `df_copy`, I get below results:
``` python
print(df_copy.ix[268434943,:])
print("--------------------------")
print(df_copy.ix[268434944,:])
```
```
base       268434943
A                 78
twice      536869886
triple    805304829
Name: 268434943, dtype: int64
--------------------------
base                     93
A                         0
twice                     0
triple    3940649673949204
Name: 268434944, dtype: int64
```
As you can see, data is not identical at 268,434,944th index. This data error continues to show in the subsequent rows

Below are the versions I am using:
```
python vesrion:  3.5.2 |Anaconda 4.2.0 (64-bit)| (default, Jul  2 2016, 17:53:06) 
[GCC 4.4.7 20120313 (Red Hat 4.4.7-1)]
feather version:  0.4.0
pandas veresion:  0.18.1
```

**Comment by wesm** [comment]
Most definitely an int32 overflow someplace. Could you also show the pyarrow version `pyarrow.__version__`?

**Comment by jhskim89** [comment]
I am seeing 0.7.1

**Comment by wesm** [comment]
Reported upstream as https://issues.apache.org/jira/browse/ARROW-1756

**Comment by jhskim89** [comment]
Thank you. Not sure if this will be helpful at all, but I did not see this issue when writing and reading as parquet (i.e. using `pyarrow.Table.from_pandas`, `pyarrow.parquet.write_table`,  `pyarrow.parquet.read_table`)

**Comment by wesm** [comment]
That's helpful info. I think the overflow is probably in the Feather implementation itself. I'll take a look and hopefully won't be too hard to fix

**Comment by miguelroboso** [comment]
Hmm same here. I have a data table with 2 billion rows. R writes it to file through feather, but I cannot read it back with python.

**Comment by russellpierce** [comment]
The upstream issue in Arrow has been marked as resolved, what needs to be done to resolve it for feather (if anything)?  https://issues.apache.org/jira/browse/ARROW-1756

**Comment by wesm** [comment]
You'll want to update to pyarrow 0.8.0 as soon as it's available in pip or conda in the next few days (we just completed the Apache release vote)

**Comment by chackoge** [comment]
I have a pandas dataframe with 4.7 million rows and 234 columns (~14 Gb). In trying to write it to feather-format, I get the following error.

**Comment by wesm** [comment]
I think that's a different error, I opened https://issues.apache.org/jira/browse/ARROW-3058 about at least making the error message better. The Feather format has some underlying limitations for very large data frames -- these limitations can be fixed but a pre-requisite is being able to ship R bindings for Apache Arrow. That work is under way but it will be some time off yet

cc @hadley @romainfrancois

**Comment by wesm** [comment]
This should definitely be fixed by Feather V2, coming soon in Arrow 0.17.0
